gobblin-runtime/src/main/java/org/apache/gobblin/runtime/JobState.java

```diff
@@ -788,12 +798,60 @@ public int getJobFailures() {
       return Integer.parseInt(super.getProp(ConfigurationKeys.JOB_FAILURES_KEY));
     }
 
+    /**
+     * Computes and stores the overall data quality status based on task-level policy results.
+     * The status will be "PASSED" if all tasks passed their quality checks, "FAILED" otherwise.
+     */
+    public void computeAndStoreQualityStatus(JobState jobState) {
+      TaskLevelPolicyChecker.DataQualityStatus jobDataQuality = TaskLevelPolicyChecker.DataQualityStatus.PASSED;
+
+      for (TaskState taskState : getTaskStates()) {
+        String qualityResult = taskState.getProp(TaskLevelPolicyChecker.TASK_LEVEL_POLICY_RESULT_KEY);
+        log.info("Data quality status of this task is: " + qualityResult);
+        if (qualityResult != null && !TaskLevelPolicyChecker.DataQualityStatus.PASSED.name().equals(qualityResult)) {
```

Comment:
it would be better to get the `qualityResult` in an enum and then compare it with `DataQualityStatus.PASSED`